Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM rust:1.43.1 as build
+ENV PKG_CONFIG_ALLOW_CROSS=1
+
+WORKDIR /usr/src/labrinth
+# Download and compile deps
+COPY Cargo.toml .
+COPY Cargo.lock .
+COPY docker_utils/dummy.rs .
+# Change temporarely the path of the code
+RUN sed -i 's|src/main.rs|dummy.rs|' Cargo.toml
+# Build only deps
+RUN cargo build --release
+# Now return the file back to normal
+RUN sed -i 's|dummy.rs|src/main.rs|' Cargo.toml
+
+# Copy everything
+COPY . .
+# Build our code
+ARG SQLX_OFFLINE=true
+RUN cargo build --release
```

Comment:
why do we build twice? Just yeet the code in and then download dependencies

Reply:
It is made like this to allow for caching. The deps will only be donwloaded and compiled if there is a change in Cargo.toml, Cargo.lock or dummy.rs.
It can easily speed up build times.